go/libraries/doltcore/table/editor/index_editor.go

```diff
@@ -390,10 +387,91 @@ func (ie *IndexEditor) Index() schema.Index {
 	return ie.idx
 }
 
-// Close ensures that all goroutines that may be open are properly disposed of. Attempting to call any other function
-// on this editor after calling this function is undefined behavior.
+// StatementStarted is analogous to the TableEditor implementation, but specific to the IndexEditor.
+func (ie *IndexEditor) StatementStarted(ctx context.Context) {
+	ie.flushMutex.Lock()
```

Comment:
defer ie.flushMutex.Unlock() is more idiomatic rather than putting at the end of the function.